controllers/feeds/getFeedsV2.js

```diff
@@ -83,6 +84,13 @@ const isValidActivity = async (item, conditions) => {
     return false;
   }
 
+  // filter by threshold
+  const threshold = ACTIVITY_THRESHOLD[feed.toUpperCase()];
+  if ((item.final_score || 0) < threshold) {
+    console.log(`final_score under threshold => `, item.final_score);
+    return false;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code assumes that `ACTIVITY_THRESHOLD[feed.toUpperCase()]` will always return a valid threshold value. However, if the feed type is not defined in the `ACTIVITY_THRESHOLD` object, this line will return `undefined`, and comparing `undefined` with a number using the `<` operator will result in `false`. This could lead to unexpected behavior if new feed types are introduced in the future. To avoid potential issues, you should add a check to ensure that a valid threshold value is retrieved.

```diff
+  const threshold = ACTIVITY_THRESHOLD[feed.toUpperCase()] || 0;
-  const threshold = ACTIVITY_THRESHOLD[feed.toUpperCase()];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->